### PR TITLE
Merging deconvolution examples.

### PR DESCRIPTION
**Pull request by alexandrejaguar**
## Description
Merge two deconvolution (Wiener and Richardson-Lucy) examples.

## References
Related to issue #2506.
[1] http://scikit-image.org/docs/dev/auto_examples/filters/plot_restoration.html#sphx-glr-auto-examples-filters-plot-restoration-py
[2] http://scikit-image.org/docs/dev/auto_examples/filters/plot_deconvolution.html#sphx-glr-auto-examples-filters-plot-deconvolution-py

### CODE REVIEW

File: doc/examples/filters/plot_deconvolution.py [hunk 1]
```diff
@@ -2,54 +2,113 @@
...
+The unsupervised Wiener algorithm has self-tuned regularization
+parameters based on data learning. This is not common and based on the
+following publication. The algorithm is based on a iterative Gibbs
+sampler that draw alternatively samples of posterior conditional law of
+the image, the noise power and the image frequency power.
+
+.. [1] François Orieux, Jean-François Giovannelli, and Thomas
+       Rodet, "Bayesian estimation of regularization and point
+       spread function parameters for Wiener-Hunt deconvolution",
+       J. Opt. Soc. Am. A 27, 1593-1607 (2010)
```
Comment (sciunto) [inline doc/examples/filters/plot_deconvolution.py]: Please, add `DOI:10.1364/JOSAA.27.001593`

Comment (alexandrejaguar) [inline doc/examples/filters/plot_deconvolution.py]: @sciunto you got it.

File: doc/examples/filters/plot_deconvolution.py [hunk 2]
```diff
@@ -2,54 +2,114 @@
...
+--------------------------------------
+First a noisy version of an image is deconvolved using Wiener
+and unsupervised Wiener algorithms. These algorithms are based on
+linear models that can't restore sharp edges as much as non-linear
+methods (like TV restoration), but are much faster.
 
-.. [2] http://en.wikipedia.org/wiki/Richardson%E2%80%93Lucy_deconvolution
+The Wiener filter is an inverse filter based on a PSF (Point Spread
+Function), the prior regularization (penalization of high frequency)
+and the tradeoff between the data and prior adequacy. The regularization
```
Comment (stefanv) [inline doc/examples/filters/plot_deconvolution.py]: What does "prior adequacy" mean?

File: doc/examples/filters/plot_deconvolution.py [hunk 3]
```diff
@@ -2,54 +2,114 @@
...
+parameter must be hand tuned.
+
+The unsupervised Wiener algorithm has self-tuned regularization
+parameters based on data learning. This is not common and based on the
```
Comment (stefanv) [inline doc/examples/filters/plot_deconvolution.py]: What is "data learning"?

**Comment by stefanv** [comment]
Looks good, just some minor requests for clarification.

File: doc/examples/filters/plot_deconvolution.py (hunk 2 above)
Comment (alexandrejaguar) [inline doc/examples/filters/plot_deconvolution.py]: @stefanv please see comment below.

File: doc/examples/filters/plot_deconvolution.py (hunk 3 above)
Comment (alexandrejaguar) [inline doc/examples/filters/plot_deconvolution.py]: @stefanv I'd need to say, I don't know. I only joined the documentation, and I don't know what the original author meant. I would need to research a little before asking these questions and improving the documentation.

**Comment by alexandrejaguar** [comment]
@stefanv please check the corrections.

**Comment by stefanv** [comment]
Thanks, @alexandrejaguar, this looks great; I have one question, and I saw Travis failed (probably not related).

**Comment by alexandrejaguar** [comment]
Hi @stefanv, I couldn't find your question :)
About Travis, apparently it fails for the Python 2 build. I wonder why...

**Comment by soupault** [comment]
@alexandrejaguar looks pretty clear to me:
```
  File "doc/examples/filters/plot_deconvolution.py", line 27
SyntaxError: Non-ASCII character '\xc3' in file doc/examples/filters/plot_deconvolution.py on line 28, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details
```

**Comment by alexandrejaguar** [comment]
@soupault OK, let me check if I can solve that.

File: doc/examples/filters/plot_deconvolution.py [hunk 4]
```diff
@@ -1,55 +1,117 @@
...
+Wiener and unsupervised Wiener filters
+--------------------------------------
+First, a noisy version of an image is deconvolved using Wiener
+and unsupervised Wiener algorithms. These algorithms are based on
+linear models that cannot restore sharp edges as much as non-linear
+methods (like TV restoration), but are faster.
 
-.. [2] http://en.wikipedia.org/wiki/Richardson%E2%80%93Lucy_deconvolution
+The Wiener deconvolution algorithm is an inverse filter based on an
+inpulse response function (also named point-spread function, or PSF), a
```
Comment (soupault) [inline doc/examples/filters/plot_deconvolution.py]: inpulse -> impulse
also named -> also called

File: doc/examples/filters/plot_deconvolution.py [hunk 5]
```diff
@@ -1,55 +1,117 @@
...
+filter to penalize the restored image frequencies (regularization), and
+a parameter which tunes the balance between the data and the
+regularization.
+
+The unsupervised Wiener algorithm has self-tuned regularization
+parameters based on the Gibbs sampling algorithm. The iterative Gibbs
+sampler draws alternatively samples of posterior conditional law of
```
Comment (soupault) [inline doc/examples/filters/plot_deconvolution.py]: samples of posterior -> samples from posterior?

File: doc/examples/filters/plot_deconvolution.py [hunk 6]
```diff
@@ -1,55 +1,118 @@
+#!/usr/bin/python
```
Comment (stefanv) [inline doc/examples/filters/plot_deconvolution.py]: Is this used in any of the other examples?  If so, should probably be `/usr/bin/env python`.

Comment (jni) [reply doc/examples/filters/plot_deconvolution.py]: *probably ➡️ definitely! However I'm happy to get this in as-is and fix later. But currently we need a rebase or cmd-line merge.

Comment (jni) [reply doc/examples/filters/plot_deconvolution.py]: Oh and actually it should just be removed — it's not used anywhere else.

**Comment by alexandrejaguar** [comment]
Dear all,
I'm truly sorry I'm not working here in a while, life is running me over!
What do we need to merge this one?
Thanks!

File: doc/examples/filters/plot_deconvolution.py [hunk 7]
```diff
@@ -1,55 +1,118 @@
...
+# -*- coding: utf-8 -*-
```
Comment (sciunto) [inline doc/examples/filters/plot_deconvolution.py]: Useless as well.

**Comment by alexandrejaguar** [comment]
Hi @sciunto, could you help me here please?
I'm getting this error when rebasing:

    jaguar@spaghetti:~/Documents/Repositories/scikit-image$ git merge --no-ff examples_deconvolve 
    CONFLICT (modify/delete): doc/examples/filters/plot_restoration.py deleted in examples_deconvolve 
    and modified in HEAD. Version HEAD of doc/examples/filters/plot_restoration.py left in tree.
    Auto-merging doc/examples/filters/plot_deconvolution.py
    CONFLICT (content): Merge conflict in doc/examples/filters/plot_deconvolution.py
    Automatic merge failed; fix conflicts and then commit the result.

Thanks!

**Comment by sciunto** [comment]
You have to edit the file to solve the conflict, add it and continue the rebase process. Please, read doc/source/gitwash/development_workflow.rst and the pointers, it will help.

**Pull request closed without merging**